hyppo/independence/friedman_rafsky.py

```diff
@@ -0,0 +1,206 @@
+import random
+from numba import jit
+import numpy as np
+
+from .base import IndependenceTest, IndependenceTestOutput
+from ..tools import perm_test
+
+
+class FriedmanRafsky(IndependenceTest):
+    r"""
+    Friedman-Rafksy (FR) test statistic and p-value.
+    This is a multivariate extension of the Wald-Wolfowitz
+    runs test for randomness. The normal concept of a 'run'
+    is replaced by a minimum spanning tree (MST) calculated between
+    the points in respective data sets with edge weights defined
+    as the Euclidean distance between two such points. After MST
+    has been determined, all edges such that both corresponding
+    nodes do not belong to the same class are severed and the
+    number of independent resulting trees is counted. This test is
+    consistent against similar tests.
```

Comment:
Add notes section discussing how to calculate the test statistic, and a references section citing the correct paper. As [here](https://github.com/neurodata/hyppo/blob/f7a1f6c3ba214884c7f508ee60159b99962a3e9c/hyppo/independence/rv.py#L19)